src/daemon/udp_connection/dlt_daemon_udp_socket.c

```diff
@@ -0,0 +1,262 @@
+    {
+        return ret_val;
+    }
+
+    if(DLT_RETURN_OK != dlt_daemon_udp_socket_open(&fd, daemon_local->flags.port)){
+        dlt_log(LOG_ERR, "Could not initialize udp socket.\n");
+    }
+    else{
+        // assign to global udp fd
+        g_udp_sock_fd = fd;
+        // set global multicast addr
+        dlt_daemon_udp_setmulticast_addr(daemon_local);
+        ret_val = DLT_RETURN_TRUE;
+        dlt_log(LOG_INFO, " initialize udp socket success\n");
+    }
+
+    return ret_val;
+}
+
+// **************************************************************************
+// Function   : dlt_daemon_udp_socket_open
+// In Param   : contains udp port number
+// Out Param  : status of udp connection setup
+// Description: This funtion is used to setup DGRAM connection
+//              does socket()->bind() on udp port
+// **************************************************************************
+DltReturnValue dlt_daemon_udp_socket_open(int *sock, unsigned int servPort)
+{
+    int enable_reuse_addr = 1;
+    int sockbuffer = DLT_DAEMON_RCVBUFSIZESOCK;
+    char portnumbuffer[SOCKPORT_MAX_LEN] = {0};
+    struct addrinfo hints, *servinfo = NULL, *addrinfo_iterator = NULL;
+    int getaddrinfo_errorcode = -1;
+
+    if(sock == NULL)
+    {
+        return DLT_RETURN_ERROR;
+    }
+
+    memset(&hints, 0, sizeof hints);
+#ifdef DLT_USE_IPv6
+    hints.ai_family = AF_INET6; // force IPv6 - will still work with IPv4
+#else
+    hints.ai_family = AF_INET;
+#endif
+    hints.ai_socktype = SOCK_DGRAM;// UDP Connection
+    hints.ai_flags = AI_PASSIVE; // use my IP address
+
+    snprintf(portnumbuffer, SOCKPORT_MAX_LEN, "%d", servPort);
+    if ((getaddrinfo_errorcode = getaddrinfo(NULL, portnumbuffer, &hints, &servinfo)) != 0) {
+        dlt_vlog(LOG_WARNING, "[%s/%d] getaddrinfo: %s\n", __func__, __LINE__,
+                gai_strerror(getaddrinfo_errorcode));
+        return DLT_RETURN_ERROR;
+    }
+
+    for(addrinfo_iterator = servinfo; addrinfo_iterator != NULL; addrinfo_iterator = addrinfo_iterator->ai_next) {
+        if ((*sock = socket(addrinfo_iterator->ai_family, addrinfo_iterator->ai_socktype,
+                     addrinfo_iterator->ai_protocol)) == SYSTEM_CALL_ERROR) {
+            dlt_log(LOG_WARNING, "socket() error\n");
+            continue;
+        }
+        dlt_vlog(LOG_INFO,
+                "[%s/%d] Socket created - socket_family:%i socket_type:%i, protocol:%i\n",
+                __func__, __LINE__,addrinfo_iterator->ai_family,
+                addrinfo_iterator->ai_socktype, addrinfo_iterator->ai_protocol);
+
+        if (setsockopt(*sock, SOL_SOCKET, SO_REUSEADDR, &enable_reuse_addr, sizeof(int))
+                       == SYSTEM_CALL_ERROR)
+        {
+            dlt_vlog(LOG_WARNING, "[%s/%d] Setsockopt error %s\n", __func__, __LINE__,
+                    strerror(errno));
+            continue;
+        }
+
+        if(setsockopt(*sock, SOL_SOCKET, SO_RCVBUF, &sockbuffer, DLT_DAEMON_RCVBUFSIZESOCK)
```

Comment:
The last parameter is sizeof(sockbuffer) or sizeof(int), not DLT_DAEMON_RCVBUFSIZESOCK.

Reply:
Done